### ISSUE TITLE
Helm swallowing `dired-do-find-regexp-and-replace` output

### ISSUE THREAD

**Issue by Cbeck527** [syl20bnr/spacemacs#10542]
Hi! Huge spacemacs fan, I really appreciate what y'all are doing for the community. You've probably heard this before, but spacemacs was my gateway drug to more powerful emacs features, and I'm really grateful for it! ❤️ 

#### Description :octocat:
`dired-do-find-regexp-and-replace` output is being captured by helm, and there aren't any interactive options to actually change the matched text.

#### Reproduction guide :beetle:
- Start Emacs
- Open dired of directory with text files
- Mark files
- `Q` for `dired-do-find-regexp-and-replace`
- type query, `RET`

*Observed behaviour:* :eyes: :broken_heart:
- Helm minibuffer opens with list of results, without ability to replace
- Press `RET` on candidate and it opens the file in a buffer with 'No suitable matches here'

*Expected behaviour:* :heart: :smile:
- Minibuffer opens with options to replace match, replace all matches, etc

#### System Info :computer:
- OS: darwin
- Emacs: 25.3.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0fa3658c)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(csv sql ansible
     (auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'complete)
     dash deft docker emacs-lisp git github
     (go :variables go-tab-width 4 gofmt-command "goimports")
     html helm javascript
     (markdown :variables
               '(markdown-pre-face
                 ((t
                   (:foreground "#586e75")))))
     nginx
     (org :variables org-enable-bootstrap-support t org-enable-github-support t org-enable-reveal-js-support t org-projectile-file "TODOs.org")
     osx
     (python :variables python-test-runner
             '(pytest nose))
     (ruby :variables ruby-enable-enh-ruby-mode t ruby-version-manager 'rbenv ruby-test-runner 'rspec)
     shell-scripts
     (spell-checking :variables spell-checking-enable-by-default nil)
     syntax-checking swift terraform
     (version-control :variables version-control-global-margin t version-control-diff-tool 'diff-hl)
     yaml)
```
- System configuration features: NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS

**Comment by CeleritasCelery** [comment]
can you reproduce with stock Emacs and helm using [these](https://github.com/emacs-helm/helm/wiki/Bugs) instructions?

**Comment by tavurth** [comment]
I've been having the same issue:
`0.300.0@27.0.50 (spacemacs@develop)`
```
emacs --version
GNU Emacs 27.0.50
Copyright (C) 2018 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
```

My workaround for this is to use `projectile-replace` by pressing `C-c p r` or `M-m p R`, while this works well for the project level, it doesn't support replacing in a directory (as far as I know.)

**Comment by Cbeck527** [comment]
@CeleritasCelery using `emacs-helm.sh` everything behaves as expected

**Comment by CeleritasCelery** [comment]
I am not master and am not seeing this issue. `dired-do-find-regexp-and-replace` works fine for me.

**Comment by tavurth** [comment]
Directory containing any files with dired:
![screenshot from 2018-04-10 11-37-20](https://user-images.githubusercontent.com/100964/38545820-9fe96ba4-3cb3-11e8-8447-376877c09bce.png)

Selecting all files in the dired buffer
![screenshot from 2018-04-10 11-37-24](https://user-images.githubusercontent.com/100964/38545819-9fc211b2-3cb3-11e8-8ec3-e83495e08385.png)

`Q` perform query replace in selected buffers
![screenshot from 2018-04-10 11-37-38](https://user-images.githubusercontent.com/100964/38545817-9f99ce32-3cb3-11e8-8983-d1903610b8be.png)

This window appears, when there should be a search and replace instance
![screenshot from 2018-04-10 11-37-44](https://user-images.githubusercontent.com/100964/38545816-9f73540a-3cb3-11e8-8dec-0d98cafdba31.png)

**Comment by CeleritasCelery** [comment]
![image](https://user-images.githubusercontent.com/10623982/38563071-40aeb762-3c99-11e8-8c22-6a9bea9ca7e9.png)

Still not seeing helm take over xref. Do you have [this](https://github.com/brotzeit/helm-xref) package installed by chance?

**Comment by tavurth** [comment]
It seems like I do have it installed, although I've not explicitly added it.
```
helm-pydoc 
helm-gitignore 
helm-css-scss 
helm-company 
helm-c-yasnippet 
helm-xref 
helm-themes 
helm-swoop 
helm-purpose 
helm-projectile 
helm-mode-manager 
helm-make 
helm-flx 
helm-descbinds 
helm-ag 
ace-jump-helm-line
```

```
dotspacemacs-additional-packages
'(vlf
  js-doc
  indium
  web-mode
  prettier-js
  rainbow-mode
  editorconfig
  ace-jump-mode
  multiple-cursors)
```

**Comment by tavurth** [comment]
![screenshot from 2018-04-11 15-49-07](https://user-images.githubusercontent.com/100964/38617505-e4a2d860-3d9f-11e8-9b71-085205e2529e.png)

Looks like it's coming from the `auto-completion` layer. 
I've disabled that for testing and after confirming that the `helm-xref` package has been removed, the bug still exists:
![screenshot from 2018-04-11 15-51-38](https://user-images.githubusercontent.com/100964/38617650-3f2d1bec-3da0-11e8-9ce8-9dbae33b77b3.png)

**Comment by CeleritasCelery** [comment]
That means the package is still installed. Add it to `dotspacemacs-excluded-packages` and delete it from `~/.emacs.d/elpa`. Then restart. (Also I really like your theme, which is that?)

**Comment by tavurth** [comment]
Yup, those actions have now fixed the issue, it's working as intended. Thank you!

`gotham` is the theme, it's nice, although there's two issues:
1) With unmatched closing braces, you should `describe-face` while the cursor is on one of them and change the color to something more useful.
2) With git merge conflicts, the same issue and solution

**Comment by Cbeck527** [comment]
I can confirm that removing `helm-xref` solves the problem! Thanks for the help @CeleritasCelery! I'm guessing this issue can be closed since it's a downstream package bug?

**Comment by CeleritasCelery** [comment]
Do we know which package is including it yet?

**Comment by tavurth** [comment]
The only one I could find is the standard Spacemacs layer `auto-completion`

**Comment by CeleritasCelery** [comment]
I don't see it [there](https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bcompletion/auto-completion/packages.el). Can you point out where you are seeing it?

**Comment by tavurth** [comment]
As far as I understand, [`+company` requires helm](https://github.com/syl20bnr/spacemacs/blob/0fa3658cd8e283825dcd0a54ce1579dec55eb568/layers/%2Bcompletion/auto-completion/packages.el#L21) => [which requires helm-xref](https://github.com/syl20bnr/spacemacs/blob/0fa3658cd8e283825dcd0a54ce1579dec55eb568/layers/%2Bcompletion/helm/packages.el#L28)

**Comment by CeleritasCelery** [comment]
I see that. That change is not present in master, which is why I was not seeing this issue. 

> dired-do-find-regexp-and-replace output is being captured by helm, and there aren't any interactive options to actually change the matched text.

I haven’t used `helm-xref`, but I can’t help but think there is a way to change the matches text. Otherwise this package would be pretty useless 😀.

**Comment by guraltsev** [comment]
I am having the same problem. The downstream package is very barren https://github.com/brotzeit/helm-xref sonno documentation is provided. I still believe that this should be considered a spacemacs bug since it is not even clear that helm-xref should intervene when calling dired-do-find-regexp-and-replace.

Can someone more emacs savvy maybe post a config line to make helm-xref leave dired-do-find-regexp-and-replace alone? Thanks.

**Comment by pcesar22** [comment]
Any updates to this? I also have this issue on develop.

**Comment by muusik** [comment]
I also stumbled upon this bug. The workaround for me is to use dired-do-query-replace-regexp instead of dired-do-find-regexp-and-replace. I am on develop and do use autocomplete layer.

**Comment by mrexox** [comment]
Try using this function (place it in your .emacs):
```lisp
(defun replace-all (directory from to)
  "Query replace all occuries in found files"
  (interactive "DDirectory: \nsReplace regexp: \nsTo: ")
  (find-grep-dired directory from)
  (when (yes-or-no-p "Start query replace?")
    (dired-toggle-marks)
    (dired-do-query-replace-regexp from to)))
```

It greps in the directory you select for the replace regexp and after `find` is done all you need is to type `y` and go through the query replaces over all found files.
